core/src/main/java/net/sourceforge/jnlp/runtime/JNLPPolicy.java

```diff
@@ -235,13 +234,14 @@ private Policy getPolicyFromUrl(String policyLocation) {
             try {
                 final URI policyUri;
                 if (policyLocation.startsWith("file://")) {
-                    policyUri = new URL(policyLocation).toURI();
+                    final String encoded = URLEncoder.encode(policyLocation, UTF_8.name());
+                    policyUri = new URL(encoded).toURI();
```

Comment:
This needed to be tested. For policyLocation file:///c:/temp/ows/AB/myjava.policy  new URL() throws exception java.net.MalformedURLException: no protocol: file%3A%2F%2F%2Fc%3A%2Ftemp%2Fows%2FAB%2Fmyjava.policy.

Reply:
did fix the unit test to detect malformed URL and also updated the implementation